### PR TITLE
Utilities updates

### PR DESCRIPTION
**Pull request by whodes**
Updating Utilities class to use the new `SmileIdentiyService`

# Changes 
- `getJobStatus` now returns `JobStatusResponse`
- `pollJobStatus` now returns `JobStatusResponse`
- removed methods that aren't needed 
- Updated signature to have `getInstant()` method 
- Added new method to Utils to return correct sid server 
- Added new constructor to Options class 
-  Updated fallbackAdapter to use InstantAdapter & PartnerParam adapter 
- test ✨

### CODE REVIEW

File: src/test/java/smile/identity/core/UtilitiesTest.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        assertEquals(request.getUserId(), "user");
+        assertEquals(request.isHistory(), options.isReturnHistory());
+        assertEquals(request.isImageLinks(), options.isReturnImageLinks());
+    }
+
+    @Test
+    public void pollForJobStatus() throws Exception {
+        Options options = new Options(false, false, true, true, "");
+        SmileIdentityService service = mock(SmileIdentityService.class);
+        Whitebox.setInternalState(utilities, service);
```
[the remark this answers is not in the document]
Comment (whodes) [reply src/test/java/smile/identity/core/UtilitiesTest.java:109]: It allows us to set field values of classes. So here I'm using it to set the smileIdentityService as a mock the to verify that the correct variables are being passed to the pollJobStatus method

File: src/main/java/smile/identity/core/Utilities.java [hunk 2]
```diff
@@ -1,255 +1,76 @@
...
-        this(partner_id, api_key, sid_server);
-        this.connectionTimeout = connectionTimeout;
-        this.readTimeout = readTimeout;
-    }
 
-    public String get_job_status(String user_id, String job_id, String options) throws Exception {
-        JSONObject optionsJson = null;
+    public Utilities(String partnerId, String apiKey, String sidServer) {
+    	this.partnerId = partnerId;
+        String url = Utils.getSidServer(sidServer);
```
[the remark this answers is not in the document]
Comment (whodes) [reply src/main/java/smile/identity/core/Utilities.java:21]: Utilities is the name used across all of the sdks, but I normally use the name `utils` for a file that hosts shared utility functions. Guess we can go with `Helpers` or something similar

File: src/main/java/smile/identity/core/ConfigHelpers.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    public static String getVersion() {
+        return properties.getProperty("version");
+    }
+
+    public static String getApiVersion() {
+        return properties.getProperty("apiVersion");
+    }
+
+    public static String getSidServer(String sidServer) {
+        if (sidServer.equals(TEST_ENV)) {
```
[the remark this answers is not in the document]
Comment (beastawakens) [reply src/main/java/smile/identity/core/ConfigHelpers.java:21]: We definitely want to keep open the scope for dynamic environmenting!

**Review by ayinloya — APPROVED** [review]
LGTM

**Pull request merged**